lib/opensusebasetest.pm

```diff
@@ -15,7 +15,8 @@ use IO::Socket::INET;
 sub new {
     my ($class, $args) = @_;
     my $self = $class->SUPER::new($args);
-    $self->{in_wait_boot} = 0;
+    $self->{in_wait_boot}    = 0;
+    $self->{in_boot_desktop} = 0;
```

Comment:
I don't understand what would be the difference between "in_wait_boot" and "in_boot_desktop". Basically I assume we are in the same state, i.e. "in_wait_boot", unless we are not using the "wait_boot" method.

Reply:
the problem is wait_serial, it checks for first time for shutdown, then checks boot to desktop where login prompt appears, so it doesn't work for the 2nd check, this is limitation, so I have to split two checks. Any other idea?

Reply:
To be honest, there is also third state, when we reboot, so it makes sense to check for both serial lines in that case.